websites/K/Kinopoisk/presence.ts

```diff
@@ -0,0 +1,215 @@
+const presence = new Presence({
+	clientId: "1034799018980679680",
+});
+
+let video = {
+	duration: 0,
+	currentTime: 0,
+	paused: true,
+};
+
+presence.on(
+	"iFrameData",
+	(data: { duration: number; currentTime: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			details: "Где-то на сайте",
+			smallImageText: "Kinopoisk",
+		},
+		[privacy, time] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("time"),
+		]),
+		{ hostname, pathname } = document.location,
+		textContent = (tags: string) => {
+			return document.querySelector(tags)?.textContent?.trim();
+		},
+		pageTitle = (string: string) => {
+			return document
+				.querySelector<HTMLTitleElement>("title")
+				.textContent.split(string);
+		},
+		strings = await presence.getStrings({
+			play: "general.watchingVid",
+			pause: "general.paused",
+		});
```

Comment:
Strings should be fetched like this (sorry for the long example). If you were to keep strings the way you are fetching them, it would result in thousands of requests being sent because the `UpdateData` event is emitted everytime there is a page change
https://github.com/PreMiD/Presences/blob/1f504d58d036edbbba6a3d66567090124b2129f7/websites/D/Discord/presence.ts#L4-L72
https://github.com/PreMiD/Presences/blob/1f504d58d036edbbba6a3d66567090124b2129f7/websites/D/Discord/presence.ts#L76-L77
https://github.com/PreMiD/Presences/blob/1f504d58d036edbbba6a3d66567090124b2129f7/websites/D/Discord/presence.ts#L110-L113

Reply:
> Strings should be fetched like this (sorry for the long example). If you were to keep strings the way you are fetching them, it would result in thousands of requests being sent because the `UpdateData` event is emitted everytime there is a page change
> 
> https://github.com/PreMiD/Presences/blob/1f504d58d036edbbba6a3d66567090124b2129f7/websites/D/Discord/presence.ts#L4-L72
> 
> 
> https://github.com/PreMiD/Presences/blob/1f504d58d036edbbba6a3d66567090124b2129f7/websites/D/Discord/presence.ts#L76-L77
> 
> 
> https://github.com/PreMiD/Presences/blob/1f504d58d036edbbba6a3d66567090124b2129f7/websites/D/Discord/presence.ts#L110-L113

I wanted to do like this right away, but then I saw the solution used. Next time I will know.